### DISCUSSION TITLE
Player Provider Poll #1

### DISCUSSION THREAD

**Discussion by OzGav** [music-assistant/support#1200] [Feature Polls]
When the HA integration is written all of the HA media players will become available. However,  native player support in MA will ensure that all MA features will be supported. The list below is taken from existing requests in these discussions but there are about 80 integrations in HA tagged with media player so if you want to suggest another then put it in the comments but before you do check if someone else has already asked for it and if so just upvote the comment. 

If you would like to discuss these options in more detail then here are the threads. 
Roku https://github.com/orgs/music-assistant/discussions/1038 and https://github.com/orgs/music-assistant/discussions/430
MPD https://github.com/orgs/music-assistant/discussions/1170
HEOS https://github.com/orgs/music-assistant/discussions/1167
Companion App https://github.com/orgs/music-assistant/discussions/396
Alexa https://github.com/orgs/music-assistant/discussions/431
Web Browser https://github.com/orgs/music-assistant/discussions/436
Yamaha Multicast https://github.com/orgs/music-assistant/discussions/409

**Comment by Guigui37** [comment]
Kodi

**Reply by OzGav** [reply]
My kodi already shows as an Airplay and DLNA player?

**Reply by Guigui37** [reply]
Does it works ? Because here I see it also, but does not work (as DLNA player, not tested with airplay)

**Reply by OzGav** [reply]
Actually looks like there might be an issue as MA is not seeing my Kodi box now. Lets see what happens with b28 which is due in the next 24 hours

**Reply by marcelveldt** [reply]
Kodi can also be controlled natively which I think is preferred over its dlna or airplay implementation ?

**Comment by Guigui37** [comment]
Onkyo (DLNA not working yet)

**Reply by marcelveldt** [reply]
Onkyo receivers do not have a native protocol/api for playback. They use Chromecast and Airplay.
Another alternative is to hook a simple streaming puck to it, connected to its optical port. For example Chromecast audio (if you can get your hands on one) or otherwise any squeezelite compatible device is fine.

Once the HA integration is done we are going to offer a feature where you can attach "controls" from HA to your player devices in MA. For example attach the volume control and source selection of your Onkyo receiver to a chromecast based player.

**Reply by regpro-bv** [reply]
Hi Marcel, although this thread is quite old, I do have multiple Onkyo NCP-302 and also a receiver that is using FlareConnect technology.
Is it possible to use music-assistant via this technology?

Appwise, Onkyo provided either https://play.google.com/store/apps/details?id=com.onkyo.playfi&hl=en_US or https://play.google.com/store/apps/details?id=com.onkyo.jp.onkyocontroller&hl=en_US

**Reply by OzGav** [reply]
Someone would need to step in to write this if it  is possible. There is no spare capacity for new features at the moment.

**Comment by RJ-Make** [comment]
Web browser as in "This Device"

**Reply by OzGav** [reply]
Is this a question?

**Reply by RJ-Make** [reply]
Sorry, Yes is it the web player that provides the "On This Device"?

**Reply by OzGav** [reply]
Yes that is the idea. Please see the linked discussion

**Reply by OzGav** [reply]
Just want to add that the Snapcast provider allows playback to any web browser. This is the current solution. Also the latest beta allows playback to Fully Kiosk Browsers.

**Comment by nagyrobi** [comment]
- [Linkplay](https://github.com/nagyrobi/home-assistant-custom-components-linkplay) (I've sent a device to @marcelveldt)
- Squeezebox / SlimProto

**Reply by marcelveldt** [reply]
> * [Linkplay](https://github.com/nagyrobi/home-assistant-custom-components-linkplay) (I've sent a device to @marcelveldt)
> * Squeezebox / SlimProto

True! I did not forget about that. As Linkplay is a pretty popular platform it should indeed be on the list.

**Reply by Makrokaba** [reply]
Sure? ;-) I'd also love to see Linkplay supported - especially since the universal group player does not sync them any more.

**Reply by mitrokun** [reply]
Linkplay is still in the plans? Now the most stable work with these players is possible when using DLNA. In the airplay or HA media players mode, they work unstably. 
It would be great to use the built-in multiroom mechanism.

**Reply by probert94** [reply]
I would also love to see Linkplay support. At the moment I added linkplay devices to HA and then use HA players in MA.
However, as soon as I group them, they seem to disappear from MA completely.

**Reply by OzGav** [reply]
It is normal behaviour that when a group is powered on that the child players are removed from view. Just check if they are still in the players tab of the settings.

**Reply by probert94** [reply]
@OzGav In my case not only the child players have been removed, but also the master. Now I checked again and only one of 5 players is shown, even if they are all available in Home Assistant.

**Reply by OzGav** [reply]
This not the place to have an extensive discussion about this. You should open your own Q&A

**Reply by points-unknown** [reply]
Yes please to linkplay provider. Still waiting in this

**Reply by MagicMicky** [reply]
I'm also curious about a linkplay integration - I don't have MusicAssistant yet but it is not clear so far what it means to have a native linkplay integration and rely on DLNA / airplay compared to for example spotcast targeting a linkplay device

**Reply by OzGav** [reply]
Spotcast has nothing to do with MA. Native player providers are optimised for music playback with MA and also means that HA is not required.

**Reply by MagicMicky** [reply]
@OzGav thanks for the replies. Do I understand correctly that if I integrate linkplay with ie: airplay, what actually happens is that spotify streaming happens between MA <> Spotify, and then it's communicated over airplay from MA to my players? I'm supposing that's also what would go away if linkplay was supported natively (Spotify would be directly streamed to the player natively)
(Sorry if the question is bit basic, but i couldn't really find the information looking at the docs!)

**Reply by OzGav** [reply]
This picture from the docs is designed to show what happens. All manner of music providers are inputs to MA which then outputs to all manner of players. Native support means that the output is optimised for MA (and will work if HA is not involved), it is always in the centre of the path

![](https://www.music-assistant.io/assets/MA_banner.png)

**Comment by thomail** [comment]
Samsung Multiroom (WAMxxx)

**Reply by thmedia2** [reply]
Also would love to see this, as a remark to others: there is a hacs integration that allows to bring the devices in HA

**Reply by OzGav** [reply]
Not for long it is moving to HA core this week!

**Comment by SeByDocKy** [comment]
Raspiaudio devices of course....(I even find surprizing to not find ithem in the list because when MA V1 was lunch, raspiaudio was the player used in the HA live)

**Reply by marcelveldt** [reply]
raspiaudio devices work already using slimproto (so install the squeezelite firmware on it). What is missing is esphome support which will follow as soon as MA and HA are connected again.

**Reply by SeByDocKy** [reply]
Yes sorry I should mention ESPhome flashed :) ... I got a stupid reflex to  not imagine some ESP32 flashed with something different than ESPhome :)... So I will wait this MA V2 update

**Comment by polmdm** [comment]
Hi everyone,
it would be great if it were possible to have the 3.5mm jack output of the raspberry directly as a music player so that it could be connected directly to the home audio system without having to go through additional hardware.

Apparently there is no out of the box solution on home assistant that integrates local playlists, radio and takes advantage of the 3.5mm jack. Thank you very much

**Reply by nagyrobi** [reply]
Try PiCorePlayer on the Pi, and add it as a slimproto device.

**Reply by polmdm** [reply]
Thank you, I'll give it a try. Nevertheless I would need an RPi, other the one on which I am currently running HA & MA, in the network, right?

**Reply by nagyrobi** [reply]
Maybe https://github.com/dingedan/Hassio_addons/tree/master/local_audio_player

**Reply by OzGav** [reply]
Or the squeeze lite addon

**Comment by srout1967** [comment]
Hi, when can we use MA2.0 with esphome media players ?

**Reply by OzGav** [reply]
Either when enough people vote for it here and it gets some native support or when the HA integration is written which will expose the HA media players,

**Reply by malvin84** [reply]
I really miss this in the new MA. Anyone aware of any HA integration that would expose HA mplayers already?

**Reply by OzGav** [reply]
This is on the roadmap so we ask for your patience

**Comment by sambarlick** [comment]
Is it possible for Emby support too?

**Reply by OzGav** [reply]
This is the player provider poll. You need to vote for Emby in music provider poll #2

**Reply by sambarlick** [reply]
Sorry I didn't realise there was a second poll, thanks.

**Comment by quizzical86** [comment]
HEOS followed closely by Web Browser seem to be the clear winners of the poll. Am I right in assuming that this will guide the dev team in where to focus but that you have no clue at this stage whether support could be added in the next months or rather next year?

**Reply by OzGav** [reply]
Correct. Do note the first sentence in the poll description.

**Reply by quizzical86** [reply]
Well noted thanks :-)
But from my experience with Music Assistant V1, I don't expect the Home Assistant HEOS integration to work well with MA V2.

**Reply by OzGav** [reply]
Ok then hopefully we can squash enough bugs to get capacity to look at these other player providers sooner rather than later

**Comment by jumblies** [comment]
ESPHome

**Reply by mplinuxgeek** [reply]
Installed MA specifically for use with ESPHome but found no support :( 

I want to start getting rid of Google Cast devices...

**Reply by malvin84** [reply]
According to @OzGav
"OzGav
Jun 28, 2023
Maintainer Author

This is on the roadmap so we ask for your patience" however we would humbly ask for an update pleeease :)

**Reply by OzGav** [reply]
ESP devices that will run the squeezelite firmware will work now.

Other ESP devices will be supported in the future, probably by leveraging off the HA devices but that may come at the cost of some advanced functionality. Time will tell on that one.

**Reply by mplinuxgeek** [reply]
I know, I've poked around hence how I ended up here, just putting my vote in as ESPHome audio devices are getting pretty popular.

**Reply by heibertelf** [reply]
Squeezelite is a good step, but the beauty of the ESPhome speaker is that it is also a voice assistant, and in general, the platform allows you to attach buttons, LEDs, sensors, etc. to the speaker. And here are all the possibilities for deep integration with the smart home.
In fact, the basic ideology is DIY, which means the possibility to realize what I myself have thought up and wanted to implement in the form of a homemade device without being an advanced programmer. 
So it's very cool to see such integrations as Music Assistant in HASS, and even more cool to be able to interact with homemade ESPhome devices. Really looking forward to it and hope to see it in the near future. Thanks for your efforts and good luck!

**Comment by fredamn76** [comment]
Jellyfin 🙏

**Reply by OzGav** [reply]
Sorry @fredamn76 you need to say that in the music provider poll :-) This poll is for physical players

**Reply by fredamn76** [reply]
Ok. I´m sorry.

**Comment by creatvty** [comment]
I read everywhere you are supposed to vote for ESPHOME if you want it. Am I blind or is it not on the list?

**Reply by OzGav** [reply]
List is limited to eight. Upvote the post two up from here

**Reply by creatvty** [reply]
I see, thanks

**Comment by topogio** [comment]
Bose SoundTouch/Cinemate

**Comment by pickonedev** [comment]
FULLY KIOSK BROWSER

I would like to see that MA it will work with "Fully Kiosk Browser" media_player entities. More than that, if this can play music in sync through this, it will be maximum achievement 🥇

**Reply by TTTMita** [reply]
I would love this as well. I have 11 tablets in my home, all with Fully and a Sync music through this, could be ideal.

**Reply by EchoEchoDZF** [reply]
Same, +1

**Reply by CamTot** [reply]
I voted "Web Browser" thinking at "FULLY KIOSK BROWSER" ... 💯

**Reply by TTTMita** [reply]
Same !!!!

**Reply by pickonedev** [reply]
Voting "Web Browser" it is only about "Media" feature from the web browser, from HA.
It's ok if you voted this, but to support Fully Kiosk Browser, just up my post, maybe we can get Fully into the poll as well, or maybe it will replace other from the poll

🥇

**Reply by MMDaemonD** [reply]
Oh... i made made the same vote for webbrowser...

+1 for Fully

**Reply by MMDaemonD** [reply]
Maybe @OzGav can help up put Fully into the feature poll <3

**Reply by OzGav** [reply]
Yeah sorry if I would have to start the poll again and lose all the votes. We keep an eye on the upvotes as well don’t worry!

**Reply by pickonedev** [reply]
> Yeah sorry if I would have to start the poll again and lose all the votes. We keep an eye on the upvotes as well don’t worry!

Something strange about the poll :-D If you sum all the percentages, you won't get 100%, the result is 95% :))))))

**Reply by OzGav** [reply]
Dodgy rounding... :-)

**Reply by OzGav** [reply]
A basic Fully Kiosk player is available in the latest beta 87. However, beta 87 has introduced some other nasty bugs so we are not recommending an upgrade at the moment. However if you have a test machine and would like to try out the new player and check for bugs in it that would be great!

**Reply by pickonedev** [reply]
The Fully Kiosk Player is the best feature which this addon can have. When I saw it, I didn't believe it first.
It is great, it is working, but now it has a very small desync between devices (max 1 second), and the music was played on 5 identical devices.

**Reply by pickonedev** [reply]
Thank you @OzGav <3

**Reply by OzGav** [reply]
Yes I haven't spoken to Marcel but unfortunately I doubt there is any possibility to sync the playback as Fully Kiosk isn't designed to do that.

**Comment by nagyrobi** [comment]
See https://github.com/esphome/feature-requests/issues/2456 for a nice idea using ESPHome BT Proxy...
Play audio to any bluetooth speaker anywhere in the house through this.
- multiroom
- announcements

**Comment by 77Silvermane77** [comment]
Bluesound for me pls :)

**Reply by OzGav** [reply]
https://music-assistant.io/player-support/bluesound/

**Comment by yzlnew** [comment]
Is it possible for Xiaomi speakers? I can get its entity through [hass-xiaomi-miot](https://github.com/al-one/hass-xiaomi-miot).

**Reply by yzlnew** [reply]
More details, my model is Xiaomi Sound Pro, which supports a modified DLNA (can't be auto detected), Airplay 2 (similar 3rd party player error, consistently disconnected, no sound).

**Comment by odowd** [comment]
Bose headphones (bluetooth)

**Reply by OzGav** [reply]
picoreplayer can do this https://youtu.be/OdLSWnAanw8?si=I0clTovCSdS3Yf77

**Reply by EmandM** [reply]
I might be missing something. I can't seem to see how picoreplayer integrates with music assistant in a way that can play over bluetooth to headphones?

**Reply by OzGav** [reply]
Picoreplayer appears as a squeezelite client to MA. Picoreplayer also has a function to connect to Bluetooth speakers. So MA —> Picoreplayer—> BT

**Comment by wimjanse** [comment]
Bluesound please

**Reply by Dwarlor** [reply]
Bluesound would be great! At the moment my Bluesound Pulse Flex speakers are not visible in MASS. They are however added to Home Assistant...

**Reply by iorifly** [reply]
Please, please! Alexa!!

**Reply by OzGav** [reply]
https://music-assistant.io/player-support/bluesound/

**Comment by vbl315** [comment]
By the way, there is a mistake in the voting. Yamaha's platform is called Music Cast, not Multicast. Fix it, please. I googled the keywords "Yamaha Music Cast" and couldn't find this thread for obvious reasons. 
Also, in my opinion, there is no need to separate "Yamaha Music Cast" and "Yamaha reciever", because Yamaha network receivers without Music Cast are very old, they are quite few and they are devoid of a platform as such, only basic DLNA capabilities

**Reply by OzGav** [reply]
Unfortunately to change the text would result in the vote being reset. Sorry for the inconvenience in being able to find it. Your text above will now help.

**Comment by vbl315** [comment]
If you want to understand the real number of users of a particular platform, you should look at the number of downloads and the number of ratings in the Google Play Market, for example. The data from the vote, in which 400 people were hired, is not very representative. If we compare Yamaha Music Cast app and Danon HEOS app, Yamaha Music cast app has 50% more reviews
![Polish_20240816_214619654](https://github.com/user-attachments/assets/f3d3dedf-2926-4901-9ba3-06be9dcf24c1)

**Reply by orangesunny** [reply]
This poll is not about arguing which _player_ platform has more users. According to the comments of all other participants, this seems to exist to learn how many of Music Assistant users want/need an integration for their _player_ platform that is not supported in MA yet.

**Comment by zenguru84** [comment]
does anyone use yoradio? it's nice and it has a basic integration with HA, i can control it from media player card, but not available under players in music assistant.. :(
[https://github.com/e2002/yoradio](https://github.com/e2002/yoradio )

**Reply by OzGav** [reply]
You can add this via the HA player provider. How well it works though will be dependent on the capabilities of the HA integration

**Comment by dunxd** [comment]
Airplay to AppleTV - airplay to speakers works, but not to AppleTV which is what I have connected to my surround sound receiver.

**Comment by saarea** [comment]
Unifi Power Amp please! They show up as airplay devices but streaming to them doesn't actually work.

**Comment by selbaer** [comment]
Hmm, I hoped it was aöready request but it not  seem not to be as widespread as I thought 
[https://www.wiimhome.com/](url)
Most devices supports AirPlay and DLNA but the first is not really highres and the latter does not support sync

**Reply by probert94** [reply]
As much as I know Wiim uses Linkplay under the hood ( like for example Arylic) and this is already requested. So please leave your vote there as well :)

**Reply by selbaer** [reply]
Yepp, thanks for reminding. I forgot, it's using Linkplay...getting old is no fun :grin:

**Comment by Wakkler** [comment]
Support for native Squeezebox controls would be cool!
